public/src/main/java/com/revenuecat/purchases/models/SubscriptionOption.kt

```diff
@@ -32,5 +32,35 @@ interface SubscriptionOption : Parcelable {
     val isBasePlan: Boolean
         get() = pricingPhases.size == 1
 
+    /**
+     * The recurring [PricingPhase] of the subscription.
+     * Looks for the last finite or infinite recurring price phase of the SubscriptionOption.
+     */
+    val recurringPhase: PricingPhase?
+        get() = pricingPhases.lastOrNull {
+            it.recurrenceMode == RecurrenceMode.FINITE_RECURRING ||
```

Comment:
🤔 would the base plan ever be `FINITE_RECURRING`? i'm thinking maybe for a prepaid plan... but i don't _think_ for a regular sub?

Reply:
Yeah yeah, I think this _could_ be a thing for prepaid. This _also_ might not be a necessary check 🤷‍♂️  I deleted and re-added this multiple times 🙈

Maybe its best to just remove it and always pull the last? 🤔

Reply:
or maybe `.lastOrNull.takeIf { it.recurrenceMode == RecurrenceMode.INFINITE_RECURRING }`? checking is kinda just insurance in case BillingClient returns something unexpected, which we are kinda doing in the case of a missing price too...

i don't feel strongly though.

if we remove the FINITE_RECURRING, let's add a TODO for prepaid. if we remove both, let's add a comment that we are assuming all subs' last phase is the base plan phase.